src/controller/AutoCommissioner.cpp

```diff
@@ -285,6 +285,12 @@ CHIP_ERROR AutoCommissioner::StartCommissioning(DeviceCommissioner * commissione
     mNeedsNetworkSetup =
         mCommissioneeDeviceProxy->GetSecureSession().Value()->AsSecureSession()->GetPeerAddress().GetTransportType() ==
         Transport::Type::kBle;
+    // if we have the WiFiCredentials ,we need to set up the network
+    if (mParams.GetWiFiCredentials().HasValue())
+    {
+        // only when mNeedsNetworkSetup is true can we enter stage CommissioningStage::kWiFiNetworkSetup
+        mNeedsNetworkSetup = true;
```

Comment:
why is this not `mNeedsNetworkSetup = mParams.GetWiFiCredentials().HasValue()`?

Or if we really need an equivalent, then:

```cpp
mNeedsNetworkSetup = mNeedsNetworkSetup || mParams.GetWiFiCredentials().HasValue()
```

if using the latter, a comment should explain in what conditions mNeedsNetworkSetup may already be true.

Reply:
Actually looking above, there is already an assignment (I missed it) so this is:

```cpp
const auto transportType = mCommissioneeDeviceProxy->GetSecureSession().Value()->AsSecureSession()->GetPeerAddress().GetTransportType();

mNeedsNetworkSetup = (transportType == Transport::Type::kBLE) || mParams.GetWiFiCredentials().HasValue();
```

Reply:
Thanks andy and I will try to find a way to track the method via which the commissionnee was discovered.